src/backend/empty/src/buffer.rs

```diff
@@ -1,7 +1,7 @@
 #[derive(Debug)]
 pub struct Buffer {
     /// Size of this buffer
-    pub size: u64,
+    pub(crate) size: u64,
```

Comment:
I've also added this fix here because I've realized these types are publicly exported and these fields could otherwise be modified.